File: 003_Longest_Substring_Without_Repeating_Characters/003_2.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        if s == "":
            return 0
        if len(s) == 1:
            return 1
        length = 0
        right = 1
        left = 0
        while right < len(s):
            cur = s.find(s[right], left, right)
            if cur != -1:
                # length = max(length, right - left)
                if length < right - left:
                    length = right - left
                left = cur + 1
            right += 1
        if s[len(s)-1] not in s[left:len(s)-1]:
            return max(length, len(s) - left)
        return length

File: 003_Longest_Substring_Without_Repeating_Characters/003_2.cpp
class Solution {
public:
	int lengthOfLongestSubstring(string s) {
		int length = 0, i = 0, j = 0;
		unordered_set<char> pool;
        int size = (int)s.size();
		while (j < size && i < size) {
			if (pool.find(s[j]) == pool.end()) {
				pool.insert(s[j++]);
				length = max(length, j - i);
			} else {
				pool.erase(s[i++]);
			}
		}
		return length;
	}
};